torch/csrc/jit/script/compiler.cpp

```diff
@@ -2102,6 +2216,81 @@ struct to_ir {
         "reverseComparision: unsupported NodeKind. File a bug");
   }
 
+  std::shared_ptr<MethodValue> resolveFullyQualifiedFunctionName(const std::string& fqfn, const SourceRange & range)
+  {
+    auto tokens = c10::Split(fqfn, ".");
+    AT_ASSERT(tokens.size() > 0);
+    auto prefix = environment_stack->getSugaredVar(tokens.at(0), range);
+
+    if (tokens.size() > 1)
+    {
+      for (size_t i = 1; i < tokens.size(); i++)
+      {
+        prefix = prefix->attr(range, method, tokens[i]);
+      }
+    }
+
+    auto mv = std::dynamic_pointer_cast<MethodValue>(prefix);
+    AT_ASSERT(mv);
+    return mv;
+  }
+
+  std::pair<std::shared_ptr<MethodValue>, std::string> getFullyQualifiedNameAndSchema(const Expr& expr, const std::vector<NamedValue>& inputs, const std::vector<NamedValue>& attributes)
+  {
+    std::vector<std::string> names;
+    auto suv = getFullyQualifiedName(expr, names);
+    if (!suv)
+    {
+      return std::make_pair(std::shared_ptr<MethodValue>{}, "");
+    }
+    auto fqfn = c10::Join(".", names);
+    if (auto mv = std::dynamic_pointer_cast<MethodValue>(suv))
+    {
+      auto& method = mv->getMethod();
+      try {
+        method.ensure_defined();
+      } catch (std::exception&) {
+        throw ErrorReport(expr.range())
+            << " method '" << method.name()
+            << "' is called recursively involving this call site. Recursive calls are not supported";
+      }
+      return std::make_pair (std::shared_ptr<MethodValue>{}, fqfn);
+
+    }
+    else
+    {
+      return std::make_pair (std::shared_ptr<MethodValue>{}, fqfn);
+    }
+
+    AT_ASSERT(false);
+  }
+
+  std::shared_ptr<SugaredValue> getFullyQualifiedName(const Expr& expr, std::vector<std::string>& names)
+  {
+      std::shared_ptr<SugaredValue> prefix = nullptr;
+      switch (expr.kind())
+      {
+        case TK_VAR:
+        {
+          Var var(expr);
+          names.push_back(var.name().name());
+          return environment_stack->getSugaredVar(var.name(), true);
+          break;
+        }
+        case '.':
+        {
+          auto select = Select(expr);
+          auto suv = getFullyQualifiedName(select.value(), names);
+          names.push_back(select.selector().name());
+          return (suv) ? suv->attr(select.range(), method, select.selector().name()) : nullptr;
+          break;
+        }
+        default:
```

Comment:
This PR doesn't support expressions like `modules[i].method1()` or anything else except for selectors for now 😢